src/main/java/com/stanfy/helium/server/DataRetrieveServlet.java

```diff
@@ -72,10 +86,28 @@ private void putClientResponse(HttpServletResponse resp, String jsonData) throws
   }
 
   private String getDataId(HttpServletRequest req) {
-    return req.getPathInfo().replaceFirst("/", "");
+    String path = req.getRequestURI();
+
+    if (path == null || path.isEmpty()) {
+      return null;
+    }
+
+    String dataId = null;
+    if (path.startsWith(PATH_DATA)) {
+      dataId = path.replaceFirst(PATH_DATA, "");
+      if (dataId.isEmpty()) {
+        return null;
```

Comment:
this means that requests was in form `GET http://host/data/" without _@id_
Should write this in comment.